### PR TITLE
Add Request Composer tool

### PR DESCRIPTION
**Pull request by philipwalton**

### CODE REVIEW

File: meta.yaml [hunk 1]
```diff
@@ -104,8 +109,11 @@ tech:
     name: autotrack
     url: https://github.com/googleanalytics/autotrack/
   core_reporting_api:
-    name: Core Reporting API
+    name: Analytics Reporting API
     url: https://developers.google.com/analytics/devguides/reporting/core/
```
Comment (pfrisella) [inline meta.yaml]: Should we keep this as Core Reporting API v3 ?
This URL will redirect to latest version which is v4 so I suggest changing this to /core/v3. Optionally you may want to change URL below to just /core/

File: meta.yaml [hunk 2]
```diff
@@ -78,6 +78,11 @@ projects:
   type: tool
   slug: query-explorer
   tech: [core_reporting_api, embed_api, management_api, metadata_api]
+- title: Request Composer
+  type: tool
+  status: new
+  slug: request-composer
+  tech: [core_reporting_api, embed_api, management_api, metadata_api]
```
Comment (pfrisella) [inline meta.yaml]: s/core_reporting_api/reporting_api_v4

File: src/javascript/request-composer/actions/select2-options.js [hunk 3]
```diff
@@ -0,0 +1,319 @@
...
+ * @param {Object} property A property object from accountSummaries.list.
+ * @param {Object} view A view object from accountSummaries.list.
+ * @return {Promise} A promise resolved with an array of all public metrics.
+ */
+function getMetrics(account, property, view) {
+  return metadata.getAuthenticated(account, property, view).then((columns) => {
+    return columns.allMetrics((metric, id) => {
+      return metric.status == 'PUBLIC' &&
+             // TODO(philipwalton): remove this temporary exclusion once
+             // calulated metrics can be templatized using the Management API.
```
Comment (pfrisella) [inline src/javascript/request-composer/actions/select2-options.js]: s/calulated/calculated

File: src/javascript/request-composer/actions/select2-options.js [hunk 4]
```diff
@@ -0,0 +1,319 @@
...
+ * @param {Object} view A view object from accountSummaries.list.
+ * @return {Promise} A promise resolved with an array of all public metrics.
+ */
+function getV4Metrics(account, property, view) {
+  return metadata.getAuthenticated(account, property, view).then((columns) => {
+    return columns.allMetrics((metric, id) => {
+      return metric.status == 'PUBLIC' &&
+             metric.addedInApiVersion == '4' &&
+             // TODO(philipwalton): remove this temporary exclusion once
+             // calulated metrics can be templatized using the Management API.
```
Comment (pfrisella) [inline src/javascript/request-composer/actions/select2-options.js]: s/calulated/calculated

File: meta.yaml (hunk 1 above)
Comment (philipwalton) [inline meta.yaml]: Done.

File: src/javascript/request-composer/actions/select2-options.js (hunk 4 above)
Comment (philipwalton) [inline src/javascript/request-composer/actions/select2-options.js]: Done.

File: meta.yaml (hunk 2 above)
Comment (philipwalton) [inline meta.yaml]: Done.

**Comment by philipwalton** [comment]
@googlebot our team authored these commits.

**Pull request merged**